### DISCUSSION TITLE
统信UOS(linuxArm64)环境，无法输入中文

### DISCUSSION THREAD

**Discussion by wangtao0505** [CefView/QCefView#396] [Q&A] [answered]
统信UOS(linuxArm64)环境，可以正常输入英文，复制和粘贴中文。但无法输入中文。也无法调起输入法。搜狗输入法、华宇输入法都不行。怀疑是CEF的bug，测试了CEF提供的ClientDemo。cef_binary_89.0.12+g2b76680+chromium-89.0.4389.90、[cef_binary_113.3.1+g525fa10+chromium-113.0.5672.128和cef_binary_124.3.7+g091246b+chromium-124.0.6367.201都一样的效果

**Comment by tishion** [comment]
@L-Super

**Comment by L-Super** [comment, marked as answer]
参考：https://github.com/CefView/QCefView/issues/299
UOS使用的是fcitx输入法框架，CEF未适配，导致NCW无法输入中文，OSR无法光标跟随（没记错的话）

**Reply by wangtao0505** [reply]
CEF的这个bug至今未解决吗？

**Reply by L-Super** [reply]
> CEF的这个bug至今未解决吗？

是的

**Comment by tishion** [comment]
#363

https://github.com/chromiumembedded/cef/issues/1813

**Comment by wangtao0505** [comment]
![image](https://github.com/CefView/QCefView/assets/80755425/4b1dc326-1c5a-4026-adff-bbac1982b0e2)
刚刚看了，这个版本在UOS下运行可以输入中文了

**Reply by L-Super** [reply]
> ![image](https://github.com/CefView/QCefView/assets/80755425/4b1dc326-1c5a-4026-adff-bbac1982b0e2)
> 刚刚看了，这个版本在UOS下运行可以输入中文了

是指CEF NCW模式能够输入中文？

**Reply by wangtao0505** [reply]
我就直接下载下来，运行可执行文件，就可以输入中文

**Reply by wangtao0505** [reply]
执行脚本：./cefsimple --single-process --no-sandbox --disable-gpu-compositing --url="https://www.baidu.com/"
直接执行可以输入中文，但是sudo以后就没法输入了。o(╯□╰)o

**Reply by wangtao0505** [reply]
刚刚了解到，我要打开的网页中使用了onlyoffice这个Web编辑器。

**Comment by wangtao0505** [comment]
使用当前最新代码，可以正常使用。我的不能使用是因为网页里面跨域了，设置一下允许跨域就可以啦。
  //设置允许跨越
  config.addCommandLineSwitch("disable-web-security"); // 禁用同源策略
  config.addCommandLineSwitch("disable-site-isolation-trials"); // 关闭站点隔离策略,允许跨域

**Reply by tishion** [reply]
你这一系列的回复，让我不明白现在是什么问题，以及什么可以正常使用了。

**Reply by wangtao0505** [reply]
之前是我错了。没有配置环境变量，正常运行时候加载的icudtl.dat(不知道是做什么的)不是当前目录下的，导致无法输入中文。
目前主干main的代码，在UOS系统下使用OSR模式是可以正常输入中文的。（需要注意设置LD_LIBRARY_PATH=.:$LD_LIBRARY_PATH）
